x-pack/plugins/transform/public/app/services/es_index_service.ts

```diff
@@ -18,16 +18,9 @@ export class IndexService {
     return privilege.hasAllPrivileges;
   }
 
-  async dataViewExists(savedObjectsClient: SavedObjectsClientContract, indexName: string) {
-    const response = await savedObjectsClient.find<DataView>({
-      type: 'index-pattern',
-      perPage: 1,
-      search: `"${indexName}"`,
-      searchFields: ['title'],
-      fields: ['title'],
-    });
-    const ip = response.savedObjects.find((obj) => obj.attributes.title === indexName);
-    return ip !== undefined;
+  async dataViewExists(dataViewsContract: DataViewsContract, indexName: string) {
+    const dv = (await dataViewsContract.find(indexName)).find(({ title }) => title === indexName);
+    return dv !== undefined;
```

Comment:
```suggestion
    return (await dataViewsContract.find(indexName)).some(({ title }) => title === indexName);
```

Reply:
Done in https://github.com/elastic/kibana/pull/131819/commits/36d639885f4609acb2982b083b9639e7a47b7956